internal/service/rds/cluster_activity_stream_test.go

```diff
@@ -0,0 +1,423 @@
+package rds_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/rds"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tfrds "github.com/hashicorp/terraform-provider-aws/internal/service/rds"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_rds_cluster_activity_stream", &resource.Sweeper{
+		Name: "aws_rds_cluster_activity_stream",
+		F:    func(region string) error { return nil },
+		Dependencies: []string{
+			"aws_kms_key",
+			"aws_kinesis_stream",
+			"aws_rds_cluster",
+		},
+	})
+}
+
+func TestAccAWSRDSClusterActivityStream_basic(t *testing.T) {
+	var dbCluster rds.DBCluster
+	clusterName := sdkacctest.RandomWithPrefix("tf-testacc-aurora-cluster")
+	instanceName := sdkacctest.RandomWithPrefix("tf-testacc-aurora-instance")
+	resourceName := "aws_rds_cluster_activity_stream.test"
+	rdsClusterResourceName := "aws_rds_cluster.test"
+	kmsKeyResourceName := "aws_kms_key.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, rds.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckAWSClusterActivityStreamDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSClusterActivityStreamConfig(clusterName, instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRDSClusterActivityStreamExists(resourceName, &dbCluster),
+					testAccCheckAWSRDSClusterActivityStreamAttributes(&dbCluster),
+					acctest.MatchResourceAttrRegionalARN(resourceName, "resource_arn", "rds", regexp.MustCompile("cluster:"+clusterName)),
+					resource.TestCheckResourceAttrPair(resourceName, "resource_arn", rdsClusterResourceName, "arn"),
+					resource.TestCheckResourceAttrPair(resourceName, "kms_key_id", kmsKeyResourceName, "key_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "kinesis_stream_name"),
+					resource.TestCheckResourceAttr(resourceName, "mode", rds.ActivityStreamModeAsync),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSRDSClusterActivityStream_disappears(t *testing.T) {
```

Comment:
might be worth to add a disappears test if the db is delete whether the code knows how to handle this case